File: js/nova/datastruct/list/DoubleArrayIterator.js
var DoubleArrayIterator = function () {
	this.array = nova_null;
	this.position = 0;
	
	
};

DoubleArrayIterator.prototype.DoubleArrayIterator = function (array) {
	this.this(array);
};

DoubleArrayIterator.prototype.destroy = function () {
};

DoubleArrayIterator.prototype.this = function (array) {
	this = array;
	this.reset();
};

DoubleArrayIterator.prototype.reset = function () {
	this.position = 0;
	return this;
};

DoubleArrayIterator.prototype.accessor_hasNext = function () {
	return this.array.count > this.position;
};

DoubleArrayIterator.prototype.mutator_hasNext = function () {
};

DoubleArrayIterator.prototype.accessor_next = function () {
	if (this.hasNext()) {
		return get(this.position++);
	}
	;
	return 0;
};

DoubleArrayIterator.prototype.mutator_next = function () {
};

DoubleArrayIterator.prototype.super = function () {
};

File: java/nova/datastruct/list/DoubleArrayIterator.java
package nova.datastruct.list;

import java.util.Optional;
import nova.exception.ExceptionData;
import nova.exception.Exception;
import nova.exception.DivideByZeroException;
import nova.io.NovaConsole;
import nova.primitive.number.Number;
import nova.primitive.number.NovaByte;
import nova.primitive.number.NovaShort;
import nova.primitive.number.NovaInt;
import nova.primitive.number.NovaLong;
import nova.primitive.number.NovaFloat;
import nova.primitive.number.NovaDouble;
import nova.primitive.Null;
import nova.primitive.number.Char;
import nova.primitive.Bool;
import nova.datastruct.list.NovaArray;
import nova.datastruct.list.IntArray;
import nova.datastruct.list.CharArray;
import nova.datastruct.list.DoubleArray;
import nova.datastruct.list.IntRange;
import nova.thread.Thread;
import nova.thread.async.Async;
import nova.gc.GC;
import nova.math.NovaMath;
import nova.NovaObject;
import nova.NovaString;
import nova.System;
import nova.Class;
import nova.datastruct.list.Iterator;
import nova.datastruct.list.NoSuchElementException;

public class DoubleArrayIterator extends NovaObject implements Iterator
{
	private DoubleArray array;
	
	public int position;
	
	
	
	
	public DoubleArrayIterator(DoubleArray array)
	{
		init(array);
	}
	
	public boolean accessor_hasNext()
	{
		return array.count > position;
	}
	
	private boolean mutator_hasNext()
	{
	}
	
	public double accessor_next()
	{
		if (accessor_hasNext())
		{
			return array.get(position++);
		}
		throw new NoSuchElementException();
		return 0;
	}
	
	private double mutator_next()
	{
	}
	
	public void init(DoubleArray array)
	{
		this.array = array;
		reset();
	}
	
	public Iterator reset()
	{
		position = 0;
		return this;
	}
	
}
